### PR TITLE
[Feature] Ensure transformation keys have the same number of elements

### PR DESCRIPTION
**Pull request by antoinebrl**
## Description

From the documentation and tests it is implicit that the number of in and out keys should match in the transformations. I'm proposing to enforce that through a runtime check by adding the `strict=True` when using zip. If the sequences don't have the same number of elements a `ValueError` is raised.

Let me know if you think enforcing this assumption requires some tests. Happy to take your suggestions if you have an idea on how to easily test all the calls to `zip`.

## Motivation and Context

This will ensure the keys are provided by pairs. Currently last elements of the longest list are ignored.

## Types of changes

Not sure if preventing silent errors should be considered a bug fix or a new feature to guide the users.

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds core functionality)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [X] I have read the [CONTRIBUTION](https://github.com/pytorch/rl/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*). 
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: torchrl/envs/transforms/transforms.py [hunk 1]
```diff
@@ -88,7 +88,7 @@ def new_fun(self, observation_spec):
             _specs = observation_spec._specs
             in_keys = self.in_keys
             out_keys = self.out_keys
-            for in_key, out_key in zip(in_keys, out_keys):
+            for in_key, out_key in zip(in_keys, out_keys, strict=True):
```
Comment (vmoens) [inline torchrl/envs/transforms/transforms.py:91]: We need to be compatible with python 3.8. PyTorch and TensorDict have a zip_strict that falls back onto zip(..., strict=True) for appropriate python versions

**Review by vmoens — COMMENTED** [review]
It's a great feature! Just need to make it work with earlier versions of python

File: torchrl/envs/transforms/transforms.py (hunk 1 above)
Comment (antoinebrl) [reply torchrl/envs/transforms/transforms.py:91]: Thanks for the pointer, I will make use of it!
Py3.8 is reaching end of life at the end of the month: https://devguide.python.org/versions/. What is the plan for supporting or dropping this version in the next torch and torchrl releases?

Comment (vmoens) [reply torchrl/envs/transforms/transforms.py:91]: I think this is a 3.10 feature so we have still a good 6 months until we can fully deprecate zip_strict

Comment (antoinebrl) [reply torchrl/envs/transforms/transforms.py]: I saw there was already some imports of private functions from `tensordict` so I did it too for `_zip_strict`.

**Review by vmoens — APPROVED** [review]
LGTM thanks!

**Pull request merged**